### PR TITLE
 Small fixes to the refactoring of the streams

### PR DESCRIPTION
**Pull request by maritelle**
I am not sure in the fixe in Ping class.

### CODE REVIEW

File: src/main/java/org/red5/server/net/rtmp/event/Ping.java [hunk 1]
```diff
@@ -296,7 +296,7 @@ public void writeExternal(ObjectOutput out) throws IOException {
 		switch (eventType) {
 			case PING_CLIENT:
 			case PONG_SERVER:
-				out.writeDouble(value2.intValue());
+				out.writeInt(value2.intValue());
```
Comment (mondain) [inline src/main/java/org/red5/server/net/rtmp/event/Ping.java:299]: The ping and pong types expect an int per the spec; they are the only types handled by our Ping that require this for value 2.

File: src/main/java/org/red5/server/net/rtmp/RTMPConnection.java [hunk 2]
```diff
@@ -673,7 +673,7 @@ public boolean isValidStreamId(Number streamId) {
 		if (log.isTraceEnabled()) {
 			log.trace("Checking validation for streamId {}; reservedStreams: {}; streams: {}, connection: {}", new Object[] { d, reservedStreams, streams, sessionId });
 		}
-		if (d < 0 || !reservedStreams.contains(d)) {
+		if (d <= 0 || !reservedStreams.contains(d)) {
```
Comment (mondain) [inline src/main/java/org/red5/server/net/rtmp/RTMPConnection.java:676]: Ensure that the equals does not cause issues with a unit test, I think there was a reason for not allowing 0 here.

File: src/main/java/org/red5/server/net/rtmp/RTMPConnection.java [hunk 3]
```diff
@@ -644,7 +644,7 @@ public Number reserveStreamId() {
 			}
 		}
 
-		if (d == MAX_RESERVED_STREAMS) {
+		if ((int) d == MAX_RESERVED_STREAMS) {
```
Comment (mondain) [inline src/main/java/org/red5/server/net/rtmp/RTMPConnection.java]: If d is a Number use Number.intValue() in this type of situation, but I think a better fix is to change the type of MAX_RESERVED_STREAMS to a double instead.

File: src/main/java/org/red5/server/net/rtmp/RTMPConnection.java [hunk 4]
```diff
@@ -632,7 +632,7 @@ public void closeChannel(int channelId) {
 	}
 	
 	public Map<Number, IClientStream> getStreamsMap() {
-		return streams;
+		return Collections.unmodifiableMap(streams);
```
Comment (mondain) [inline src/main/java/org/red5/server/net/rtmp/RTMPConnection.java:635]: Yes, this is excellent.

File: src/main/java/org/red5/server/net/rtmp/RTMPConnection.java (hunk 2 above)
Comment (maritelle) [inline src/main/java/org/red5/server/net/rtmp/RTMPConnection.java:676]: There is no problems with unit tests. In other paces we consider that stream id is valid if it is more than 0.

File: src/main/java/org/red5/server/net/rtmp/event/Ping.java (hunk 1 above)
Comment (maritelle) [inline src/main/java/org/red5/server/net/rtmp/event/Ping.java:299]: Excuse me, can you explain this in other words? Is it wrong or right code? :)

Comment (mondain) [inline src/main/java/org/red5/server/net/rtmp/event/Ping.java:299]: It has to stay as an integer, the original code is already correct.

**Pull request merged**